easyrsa3/easyrsa

```diff
@@ -514,9 +514,20 @@ Your newly created PKI dir is: $EASYRSA_PKI
 
 hide_read_pass()
 {
-	(stty -echo 2>/dev/null) || set +o echo
-	read -r "$@"
-	(stty echo 2>/dev/null) || set -o echo
+	# shellcheck disable=SC2039
+	if stty -echo 2>/dev/null; then
+		read -r "$@"
+		stty echo
+	elif (set +o echo 2>/dev/null); then
+		set +o echo
+		read -r "$@"
+		set -o echo
+	elif (echo | read -r -s 2>/dev/null) ; then
```

Comment:
Ops... extra space before ';'

Reply:
`read -r -s` is a bashism, is there any case where `stty -echo` will not work but this will ?

Reply:
As I mentioned in commit log, OpenWrt:

```
$ grep PRETT /etc/os-release 
PRETTY_NAME="OpenWrt 18.06.1"
$ echo $SHELL
/bin/ash
$ stty
-ash: stty: not found
$ set -o echo # this would abort on non-interactive shell
-ash: set: illegal option -o echo
$ read -s # now I type test without echoing 
$ echo $REPLY
test
```

And there might be other embed systems that uses ash and does not have stty.

Reply:
OK, thanks for clarifying.